projects/step-core/src/lib/directives/dialog-close.directive.ts

```diff
@@ -0,0 +1,14 @@
+import { Directive, HostListener, inject } from '@angular/core';
```

Comment:
This directive copies `mat-dialog-close` functionality.
Remove his directive and replace it's usages with `mat-dialog-close`

Reply:
this is for a reason, the mat-dialog-close directive closes the dialog with an empty string instead of undefined (undefined is returned when the dialog is closed by clicking outside) and that is first of all inconsistent and second its incorrect in my opinion for every dialog to return an empty string (for example we have this dialog to edit text and it doesnt make sense for it to return an empty string upon cancel, so we want the dialog to close with undefined which we can check against and be sure that the string that gets emitted is the actual value the user intended)

Reply:
https://github.com/exense/step-frontend/pull/636/files#diff-bc3e4f53d59cb8ba11bed91ac8a33b73f0d0c07e9bbe66957b1a9e2ba5513d8dR47

Reply:
in `dialogs.service.ts`
<img width="549" alt="image" src="https://github.com/exense/step-frontend/assets/8954749/7424e64c-8123-4a38-9793-54b1d95f9d14">

Reply:
@bojidaryovchev 
In case if you need the close result other than empty string, use:
`[matDialogClose]="undefined"`

Reply:
ah, spot on, my bad, didnt think to check the docs -_-